beacon-chain/sync/initial-sync/sync_state.go

```diff
@@ -0,0 +1,110 @@
+package initialsync
+
+import (
+	"context"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	"github.com/prysmaticlabs/prysm/shared/p2p"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"go.opencensus.io/trace"
+)
+
+func (s *InitialSync) processState(msg p2p.Message) {
+	ctx, span := trace.StartSpan(msg.Ctx, "beacon-chain.sync.initial-sync.processState")
+	defer span.End()
+	data := msg.Data.(*pb.BeaconStateResponse)
+	finalizedState := data.FinalizedState
+	justifiedState := data.JustifiedState
+	canonicalState := data.CanonicalState
+	recState.Inc()
+
+	if err := s.db.SaveFinalizedState(finalizedState); err != nil {
```

Comment:
I wonder if we can use a new db method to process all these different states and blocks, we dont need to do multiple writes for this, when it can be done in a single one.